File: JavaScript/[1]Programmers/포켓몬.js
function solution(nums) {

    let answer = 0;
    let setNum = new Set(nums);
    let setNums = [...setNum];

    let pickCnt = (nums.length / 2);

    if(setNums.length < pickCnt) {
        answer = setNums.length;
    } else {
        answer = pickCnt;
    }
    return answer;
}


let nums = [3,1,2,3];
console.log(solution(nums));

File: python/[1]Programmers/코딩테스트 고득점 Kit/포켓몬.py
def solution(nums):
    answer = 0
    setNum = set(nums)
    pickCnt = int(len(nums)/2)

    if len(setNum) < pickCnt :
        answer = len(setNum)
    else :
        answer = pickCnt
    return answer
    
nums1 = [3,1,2,3] #2
nums2 = [3,3,3,2,2,4] #3
nums3 = [3,3,3,2,2,2] #2

print(solution(nums1))
print(solution(nums2))
print(solution(nums3))
